File: Java/T4/codigos2021/T04_Dialogos/app/src/main/java/com/borja/t04_dialogos/dialogos/DialogoMultiple.java
package com.borja.t04_dialogos.dialogos;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.borja.t04_dialogos.R;
import com.borja.t04_dialogos.utils.Equipo;

import java.util.ArrayList;

public class DialogoMultiple extends DialogFragment {


    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setTitle("Diálogo seleccion multiple");

        ArrayList<Equipo> listaEquipos = new ArrayList<>();
        listaEquipos.add(new Equipo("Barcelona","LaLiga", R.drawable.barsa));
        listaEquipos.add(new Equipo("Madrid","LaLiga",R.drawable.madrid));
        listaEquipos.add(new Equipo("Atleti","LaLiga",R.drawable.atleti));
        boolean[]seleccionados = {false,true,true};

        String[] equiposNombre = {"Barsa","Madrid","Atleti"};

        builder.setMultiChoiceItems(equiposNombre, seleccionados, new DialogInterface.OnMultiChoiceClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i, boolean b) {

            }
        });

        /*builder.setPositiveButton("Comunicar", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {

            }
        });*/


        return builder.create();
    }
}

File: Kotlin/T5/codigos2223/T5_Dialogos/app/src/main/java/com/develop/t5_dialogos/dialogs/DialogoMultiple.kt
package com.develop.t5_dialogos.dialogs

import android.app.Dialog
import android.content.DialogInterface
import android.content.DialogInterface.OnClickListener
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class DialogoMultiple : DialogFragment(), OnClickListener{

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireContext())
        val elementos = arrayOf("Opcion 1", "Opcion 2", "Opcion 3")
        //val elementosSelected = arrayOf(true,false,false)

        builder.setTitle("Selección multiple")

        builder.setMultiChoiceItems(elementos,null){ _, posicion, boolean ->
            Log.v("dialogos", "${posicion.toString()} ${boolean.toString()}")
        }

        builder.setPositiveButton("OK") { _, _ ->
            Log.v("dialogos", "realizar comunicacion")
            // se le comunique a la activity el numero de elementos que estan seleccionados
        }

        builder.setNeutralButton("Cerrar") { _, _ ->
            Log.v("dialogos", "cerrando boton")
        }




        return builder.create()
    }

    override fun onClick(p0: DialogInterface?, p1: Int) {
        TODO("Not yet implemented")
    }
}